tests/lib/rules/no-alert.js

```diff
@@ -22,14 +24,87 @@ eslintTester.addRuleTest("lib/rules/no-alert", {
         "a[o.k](1)",
         "foo.alert(foo)",
         "foo.confirm(foo)",
-        "foo.prompt(foo)"
+        "foo.prompt(foo)",
+        "function alert() {} alert();",
+        "var alert = function () {}; alert();",
+        "function foo() { var alert = bar; alert(); }",
+        "function foo(alert) { alert(); }",
+        "var alert = function() {}; function test() { alert(); }",
+        "function foo() { var alert = function() {}; function test() { alert(); } }",
+        "function confirm() {} confirm();",
+        "function prompt() {} prompt();",
+        "window[alert]();",
```

Comment:
Correct. It also covers the else branch in `getPropertyName` where the `MemberExpression` is computed but the property is not a `Literal`.